src/MPS.jl

```diff
@@ -504,18 +504,30 @@ end
 # TODO dispatch on form
 # TODO generalize to AbstractAnsatz
 function mixed_canonize!(tn::AbstractMPO, orthog_center)
+    if orthog_center isa Site
+        left = id(orthog_center) - 1
+        right = id(orthog_center) + 1
+    else
+        values = [id(site) for site in orthog_center]
+        orthog_center = Vector{Site}(orthog_center)
+
+        left, right = extrema(values) .+ (-1, 1)
+    end
+
```

Comment:
how about this? you don't need to collect `orthog_center` right?

```suggestion
    left, right = if orthog_center isa Site
        id(orthog_center) .+ (-1, 1)
    elseif orthog_center isa Vector{<:Site}
        extrema(id.(orthog_center)) .+ (-1, 1)
    else
        throw(ArgumentError("`orthog_center` must be a `Site` or a `Vector{Site}`")
    end

```

Reply:
ahh I think I did that because if `orthog_center` in the argument is something like `[Site(2), Site(3)]`, this has type of `Vector{Site{1}}` which does not enter on the constructor function which is defined for `Union{Site, Vector{Site}}`, that's why I had that `Vector{Site}(orthog_center)` part.